### PR TITLE
Panoramic controls for 360 videos or scenes

### PR DESCRIPTION
**Pull request by boyaquito**
Small class to control perspective camera in 360 degree videos or
scenes.

### CODE REVIEW

File: examples/js/controls/PanoramicControls.js [hunk 1]
```diff
@@ -0,0 +1,229 @@
+THREE.PanoramicControls = function( camera, domElement )
+{
+	this.camera = ( camera !== undefined ) ? camera : console.error( "PanoramicControls required a THREE.PerspectiveCamera in first argument." );
```
Comment (WestLangley) [inline examples/js/controls/PanoramicControls.js]: Your code assumes a perspective camera. This test allows an orthographic one.

File: examples/js/controls/PanoramicControls.js [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+	this.domElement = ( domElement !== undefined ) ? domElement : document.body;
+	this.target = new THREE.Vector3();
+
+	this.autoRotateSpeed = 1.0;
+
+	this.minPolarAngle = 0.0;// degrees
```
Comment (WestLangley) [inline examples/js/controls/PanoramicControls.js]: Orbit controls uses radians for this parameter. Do you mind using radians here for consistency?

File: examples/js/controls/PanoramicControls.js [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+	this.maxPolarAngle = 180.0;// degrees
+
+	this.deceleration = 0.88;
+	this.steps = 6.0;
```
Comment (WestLangley) [inline examples/js/controls/PanoramicControls.js]: `steps` is a mouse-sensitivity parameter in your code. See how this is done in the other controls. Perhaps there is a better name for this...

File: examples/js/controls/PanoramicControls.js [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+
+	this.zoomSpeed = 0.2;
+	this.zoomSteps = 1.0;
```
Comment (WestLangley) [inline examples/js/controls/PanoramicControls.js]: Do you really need two zoom sensitivity parameters?

File: examples/js/controls/PanoramicControls.js [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+	this.minZoom = this.camera.fov;
```
Comment (WestLangley) [inline examples/js/controls/PanoramicControls.js]: `zoom` is unitless in three.js. `fov` is in degrees. Call it `minFov`?

File: examples/js/controls/PanoramicControls.js [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+	this.maxZoom = 30.0;
+
+	this.enabledZoom = true;
```
Comment (WestLangley) [inline examples/js/controls/PanoramicControls.js]: In `OrbitControls`, this is `enableZoom`.

**Comment by WestLangley** [comment]
Nice idea, but I have some questions...

I am not sure I understand how this control is intended to be used. Do you allow zooming and dollying?

What happens if the camera position is not at the origin?

For me, the control seems to zoom in no matter the scroll direction. (OS X, Chrome).

Is an animation loop required? Compare with `OrbitControls`.

**Comment by boyaquito** [comment]
@WestLangley  

Thank you for your consideration, the detailed explanation and improvements. To answer your questions:

> I am not sure I understand how this control is intended to be used. Do you allow zooming and dollying?

The controls are intended to be used at first for basic spinning in the center of a scene. Later, I will add the possibility to move freely inside the level creation or scene.

> What happens if the camera position is not at the origin?

At the moment if the camera is not at the origin, things get broken. The target need to be child of the camera. That will be improved/updated too.

> For me, the control seems to zoom in regardless of the scroll direction. (OS X, Chrome).

I wasn't aware of that. I'll take a look at it.

> Is an animation loop required? Compare with OrbitControls.

I kept the animation loop to have the deceleration once the user stops interacting, for smoothness. It can be removed. 

I'll correct the naming you had pointed out, following orbit controls, to avoid confusions.

**Comment by Mugen87** [comment]
Closing. We can reopen this if @boyaquito is interested in implementing the mentioned improvement suggestions.

**Pull request closed without merging**